### PR TITLE
feat(modal-with-backdrop): basic modal with backdrop

### PR DESCRIPTION
**Pull request by AlonBH**

### CODE REVIEW

File: src/components/Modal/index.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  box-sizing: border-box;
+  transition: all 0.3s ease-out;
+
+  @media (min-width: 600px) {
+    width: 500px;
+    left: calc(50% - 250px);
+  }
+`;
+
+export default React.memo((props) => {
```
Comment (shirmagen) [inline src/components/Modal/index.js]: why memo? sure you need it?

Comment (AlonBH) [reply src/components/Modal/index.js]: Removed memo for now. I used it to reduce the number of times the modal renders to only when we want to show it.

File: src/Home/index.js [hunk 2]
```diff
@@ -1,17 +1,32 @@
...
 import CreateButton from '../components/CreateNew';
 import ApolloAppBar from '../components/AppBar';
 import { Row } from 'mui-flex-layout';
+import Modal from '../components/Modal';
 
 export default () => {
+  const [openedModal, setOpenedModal] = useState(false);
+
+  const openModal = () => {
+    console.log('open');
```
Comment (shirmagen) [inline src/Home/index.js:13]: remove console log

File: src/Home/index.js [hunk 3]
```diff
@@ -1,17 +1,32 @@
...
+    setOpenedModal(true);
+  };
+
+  const closeModal = () => {
+    console.log('close');
```
Comment (shirmagen) [inline src/Home/index.js:18]: remove console log

File: src/Home/index.js [hunk 4]
```diff
@@ -1,17 +1,32 @@
...
+    setOpenedModal(false);
+  };
   return (
     <Row width={'100%'} height={'100%'}>
       <ApolloAppBar />
       <Map />
       <VerticalToolBar />
-      <CreateButton />
+      <CreateButton onClick={() => openModal()}/>
```
Comment (shirmagen) [inline src/Home/index.js:26]: im working on create page so put it on another button, where you really want the feature to be

File: src/components/Modal/index.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  z-index: 500;
+  background-color: white;
+  width: 70%;
+  border: 1px solid #ccc;
+  box-shadow: 1px 1px 1px black;
+  padding: 16px;
+  left: 15%;
+  top: 30%;
+  box-sizing: border-box;
+  transition: all 0.3s ease-out;
```
Comment (shirmagen) [inline src/components/Modal/index.js:17]: the animation is ugly to my opinion :)
maybe we should delete

File: src/components/Modal/index.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  }
+`;
+
+export default (props) => {
+  return(
+    <div>
+      <Backdrop show={props.show} clicked={props.modalClosed}/>
+      <Modal 
+        style={{
+          transform: props.show ? 'translateY(0)' : 'translateY(-100vh)',
```
Comment (shirmagen) [inline src/components/Modal/index.js:31]: dont put the transform here in the middle of the component

Comment (shirmagen) [inline src/components/Modal/index.js:31]: why 100vh?

File: src/components/Modal/index.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  padding: 16px;
+  left: 15%;
+  top: 30%;
+  box-sizing: border-box;
+  transition: all 0.3s ease-out;
+
+  @media (min-width: 600px) {
+    width: 500px;
+    left: calc(50% - 250px);
+  }
```
Comment (shirmagen) [inline src/components/Modal/index.js:22]: why do you need this? you can do it without

**Comment by shirmagen** [comment]
not relevant right now

**Pull request closed without merging**